### PR TITLE
add audit checklist

### PR DESCRIPTION
**Pull request by xlc**

### CODE REVIEW

File: audit-checklist.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Audit Check List
```
Comment (kaichaosun) [inline audit-checklist.md]:
```suggestion
# Audit Check List (Draft)

** Disclaimer**: This document is still in an early stage and should not be used as the criteria when launching a valuable network. It's always necessary to introduce sufficient third-party auditing before deploying to production. 
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # Parity Asia Wiki / FAQ
 
 - [Substrate FAQ](./substrate-faq.md)
+- [Audit Checklist](./audit-checklist.md)
```
Comment (kaichaosun) [inline README.md]:
```suggestion
- [Audit Checklist (Draft)](./audit-checklist.md)
```

File: audit-checklist.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- Custom RPC
+  - Reasonable computational complexity
+  - Reasonable resource usage (memory & computation)
+  - Ones require heavy resource usages should be marked as unsafe
+- Unsigned Transaction
+  - Ensure they cannot be abused to spam the network
+    - i.e. validated before enter tx queue
+- Integration
+  - Check all the parameter values are sane
+  - Ensure there are tests to validate the dependent parameters
```
Comment (kaichaosun) [inline audit-checklist.md]:
```suggestion
  - Ensure there are tests or benchmarks to validate the dependent parameters
```

File: audit-checklist.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+- Read whitepaper and ensure it make sense
+- Read docs and ensure it matches to whitepaper
+- Read the code and ensure it matches to docs
+- Read the tests and ensure it make sense and have sufficent coverage for edge cases
+
+### Areas to check
+
+- Docs
+  - Docs for storages / events / calls / errors
+  - Ensure all invariants/assumptions are explicitly documented
```
Comment (kaichaosun) [inline audit-checklist.md:16]: Can you give an example for this one?

File: audit-checklist.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - Better to be documented as an assertion / expect in code if possible
+- Maintainability
+  - Code is reasonably understandbale
```
Comment (kaichaosun) [inline audit-checklist.md]:
```suggestion
  - Code is reasonably understandable
```

**Review by kaichaosun — APPROVED** [review]
👍  Thanks Bryan
